python-package/securexgboost/core.py

```diff
@@ -3056,10 +3201,38 @@ def sign_data(key, data, data_size):
     sig_len = ctypes.c_size_t(1024)
 
     # Sign data with key keyfile
-    _check_call(_LIB.sign_data(keyfile, data, data_size, signature, ctypes.byref(sig_len)))
+    _check_call(_LIB.sign_data_with_keyfile(keyfile, data, data_size, signature, ctypes.byref(sig_len)))
 
     # Cast the signature and sig_len back to a gRPC serializable format
     sig_len_as_int = sig_len.value
     signature = pointer_to_proto(signature, sig_len_as_int, nptype=np.uint8)
 
     return signature, sig_len_as_int
+
+def verify_enclave_signature(data, size, sig, sig_len):
+    """
+    Verify the signature returned by the enclave with nonce
+    """
+    arr = (ctypes.c_char * (size + CIPHER_NONCE_SIZE))()
+    add_to_sig_data(arr, data=data, data_size=size)
+    add_nonce_to_sig_data(arr, pos=size)
+    size = ctypes.c_size_t(len(arr))
+
+    pem_key = globals()["enclave_pk"]
+    pem_key_len = globals()["enclave_pk_size"]
+    # Verify signature
+    _check_call(_LIB.verify_signature(pem_key, pem_key_len, arr, size, sig, sig_len))
+
+    globals()["nonce_ctr"] += 1
+
+
+def create_enclave_signature(args):
```

Comment:
Is my understanding correct that this function is called to create a client-side signature over the arguments and nonce? If so, can we rename the function? The name is a bit confusing as it implies creating a signature in the enclave. Maybe we can rename to `create_client_signature`?